Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+    // not exist. otherwise set m_full_path to location.
+
+    if (boost::filesystem::is_directory(temp_dir)) {
+      m_full_path = temp_full_path;
+
+    } else {
+      throw std::invalid_argument("failed to load temp directory: " +
+                                  temp_dir.generic_string());
+    }
+  }
+
+  std::string fullPath() const { return m_full_path.generic_string(); }
+
+  ~ScopedFileHandle() {
+
+    // file is removed at end of file handle's lifetime
+    if (boost::filesystem::is_regular_file(m_full_path)) {
+      boost::filesystem::remove(m_full_path);
+    }
+  }
+
+private:
+  boost::filesystem::path m_full_path; // full path to file
+
+  // prevent heap allocation for ScopedFileHandle
+protected:
+  static void *operator new(std::size_t); // prevent heap allocation of scalar.
+  static void *operator new[](std::size_t); // prevent heap allocation of array.
+};
+
+} // namespace
+
+//---------------------------------------------------------------------
+
+class NexusGeometrySaveTest : public CxxTest::TestSuite {
+
+private:
+  std::pair<std::unique_ptr<Mantid::Geometry::ComponentInfo>,
+            std::unique_ptr<Mantid::Geometry::DetectorInfo>>
+      m_instrument;
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static NexusGeometrySaveTest *createSuite() {
+    return new NexusGeometrySaveTest();
+  }
+  static void destroySuite(NexusGeometrySaveTest *suite) { delete suite; }
+
+  NexusGeometrySaveTest() {
+
+    const Quat bankRotation(15, V3D(0, 1, 0));
+    const Quat detRotation(15, V3D(0, 1, 0));
+
+    auto instrument =
+        ComponentCreationHelper::createSimpleInstrumentWithRotation(
+            Mantid::Kernel::V3D(0, 0, -7), Mantid::Kernel::V3D(0, 0, 0),
+            Mantid::Kernel::V3D(0, 0, 10), bankRotation, detRotation);
+    instrument->setName("example-detector-bank");
+    m_instrument =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+  }
+
+  /*
+====================================================================
+
+IO PRECONDITIONS TESTS
+
+DESCRIPTION:
+
+The following tests are written to document the behaviour of the SaveInstrument
+method when a valid and invalid beamline Instrument are attempted to be saved
+out from memory to file. Included also are tests that document the behaviour
+when a valid (.nxs, .hdf5 ) or invalid output file extension is attempted to
+used.
+
+LIST:
+
+* test providing invalid path throws
+* test progress reporting
+* test false file extension throws
+* test instrument without sample throws
+* test instrument without source throws
+
+====================================================================
+*/
+
+  void test_providing_invalid_path_throws() {
+
+    ScopedFileHandle fileResource("invalid_path_to_file_test_file.hdf5");
+    std::string destinationFile = "false_directory\\" + fileResource.fullPath();
```

Comment:
call variable `badDestinationPath` and make const

Reply:
done